### PR TITLE
fix: run atleast one batch per searcher op [DET-3877]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change fixes a bug where a very small adaptive searcher, in which the first rung is less than one batch, would end up requesting to run zero batches because we decided in to round down to the nearest batch when using records or epochs. We now round down usually, but round up if the searcher requested we run a quantity less than the trial's global batch size.

## Test Plan
- [x] write a tests that fails and fix it.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: e2e_tests/tests/test_system.py [hunk 1]
```diff
@@ -504,3 +504,10 @@ def test_perform_initial_validation() -> None:
         config, conf.fixtures_path("no_op"), 1, has_zeroth_step=True
     )
     exp.assert_performed_initial_validation(exp_id)
+
+
+@pytest.mark.e2e_cpu  # type: ignore
```
Comment (rb-determined-ai) [inline e2e_tests/tests/test_system.py:509]: What code gets tested here, beyond the tiral workload sequencer?  That is: why make this an e2e test when we could make it a unit test?

File: master/go.mod [hunk 2]
```diff
@@ -8,6 +8,7 @@ require (
 	github.com/bmizerany/assert v0.0.0-20160611221934-b7ed37b82869 // indirect
 	github.com/bufbuild/buf v0.16.0
 	github.com/containerd/containerd v1.3.2 // indirect
+	github.com/davecgh/go-spew v1.1.1
```
Comment (rb-determined-ai) [inline master/go.mod:11]: unnecessary?

File: e2e_tests/tests/test_system.py (hunk 1 above)
Comment (stoksc) [reply e2e_tests/tests/test_system.py]: Agree, just swapped in a unit test.

File: master/go.mod (hunk 2 above)
Comment (stoksc) [reply master/go.mod]: Yeah, removed.

**Pull request merged**